engine/src/main/java/org/pentaho/di/trans/steps/mergejoin/MergeJoin.java

```diff
@@ -463,4 +469,63 @@ protected boolean isInputLayoutValid( RowMetaInterface row1, RowMetaInterface ro
     return true;
   }
 
+  @Override
+  public JSONObject doAction( String fieldName, StepMetaInterface stepMetaInterface, TransMeta transMeta,
+                              Trans trans, Map<String, String> queryParamToValues ) {
+    JSONObject response = new JSONObject();
+    try {
+      Method actionMethod = MergeJoin.class.getDeclaredMethod( fieldName + "Action", Map.class );
+      this.setStepMetaInterface( stepMetaInterface );
+      response = (JSONObject) actionMethod.invoke( this, queryParamToValues );
+      response.put( StepInterface.ACTION_STATUS, StepInterface.SUCCESS_RESPONSE );
+    } catch ( NoSuchMethodException | InvocationTargetException | IllegalAccessException e ) {
+      log.logError( e.getMessage() );
+      response.put( StepInterface.ACTION_STATUS, StepInterface.FAILURE_METHOD_NOT_RESPONSE );
+    }
+    return response;
+  }
+
+  private JSONObject previousKeysAction( Map<String, String> queryParams ) throws KettleException {
+    JSONObject response = new JSONObject();
+
+    String stepIndex = queryParams.get( "stepIndex" );
+    if ( StringUtils.isBlank( stepIndex ) || !StringUtils.isNumeric( stepIndex ) ) {
+      throw new KettleException( BaseMessages.getString( PKG, "MergeJoinDialog.ErrorGettingFields.DialogTitle" ) );
```

Comment:
Code will not be able to fetch the message from message.properties

Reply:
addressed the issue, please check